modules/elasticsearch-index/src/main/java/org/opencastproject/elasticsearch/index/objects/series/SeriesIndexUtils.java

```diff
@@ -122,17 +122,40 @@ public static SearchMetadataCollection toSearchMetadata(Series series) {
       metadata.addField(SeriesIndexSchema.RIGHTS_HOLDER, series.getRightsHolder(), true);
     }
 
+    if (!series.getExtendedMetadata().isEmpty()) {
+      addExtendedMetadata(metadata, series.getExtendedMetadata());
+    }
+
     if (StringUtils.trimToNull(series.getAccessPolicy()) != null) {
       metadata.addField(SeriesIndexSchema.ACCESS_POLICY, series.getAccessPolicy(), false);
       addAuthorization(metadata, series.getAccessPolicy());
     }
-
     if (series.getTheme() != null) {
       metadata.addField(SeriesIndexSchema.THEME, series.getTheme(), false);
     }
     return metadata;
   }
 
+  /**
+   * Adds extended metadata fields to the input document
+   *
+   * @param doc
+   *          the input document
+   * @param extendedMetadata
+   *          the extended metadata map
+   */
+  private static void addExtendedMetadata(SearchMetadataCollection doc, Map<String, Map<String,
+          List<String>>> extendedMetadata) {
+    for (String type: extendedMetadata.keySet()) {
+      Map<String, List<String>> extendedMetadataByType = extendedMetadata.get(type);
+      for (String name: extendedMetadataByType.keySet()) {
+        List<String> values = extendedMetadataByType.get(name);
+        String fieldName = SeriesIndexSchema.EXTENDED_METADATA_PREFIX.concat(type + "_" + name);
```

Comment:
This naming schema causes trouble if there are two extended metadata catalogs with the same type AND a property with the same inputID. Unlike with events, the `type` only seems to be the maintype of the flavor, not maintype + subtype. Basically this causes the two catalogs to have a kind of "shared" field in the index. No idea how realistic that scenario is though.

Reply:
There should never be two extended metadata catalogs with the same type for series, because this will already break in the database. The series service expects this to be unique. This is actually documented somewhere if I remember correctly. And I totally agree that this is bonkers, but I'm not touching that particular hornets nest with a stick right now.